File: schemas/helpdesk-customer-support/drizzle/ticket_attachments.ts
// ticket_attachments: files attached to ticket messages with metadata.
// See README.md for full design rationale.

import { pgTable, uuid, text, integer, timestamp, index } from "drizzle-orm/pg-core";
import { tickets } from "./tickets";
import { ticketMessages } from "./ticket_messages";
import { users } from "./users";

export const ticketAttachments = pgTable(
  "ticket_attachments",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    ticketId: uuid("ticket_id").notNull().references(() => tickets.id, { onDelete: "cascade" }),
    messageId: uuid("message_id").references(() => ticketMessages.id, { onDelete: "set null" }),
    fileName: text("file_name").notNull(),
    fileUrl: text("file_url").notNull(),
    fileSize: integer("file_size"),
    mimeType: text("mime_type"),
    uploadedBy: uuid("uploaded_by").notNull().references(() => users.id, { onDelete: "restrict" }),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_ticket_attachments_ticket_id").on(table.ticketId),
    index("idx_ticket_attachments_message_id").on(table.messageId),
  ]
);

File: schemas/helpdesk-customer-support/spacetimedb/ticket_attachments.rs
// ticket_attachments: Files attached to ticket messages with metadata.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = ticket_attachments, public)]
pub struct TicketAttachment {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub ticket_id: String, // UUID — FK → tickets.id (cascade delete)
    #[index(btree)]
    pub message_id: Option<String>, // UUID — FK → ticket_messages.id (set null)
    pub file_name: String,
    pub file_url: String,
    pub file_size: Option<i32>,
    pub mime_type: Option<String>,
    pub uploaded_by: String, // UUID — FK → users.id (restrict)
    pub created_at: Timestamp,
}
